src/ui/application.vala

```diff
@@ -27,8 +44,14 @@ public class Games.Application : Gtk.Application {
 	}
 
 	public void load_game_list () {
-		var dummy_source = new Games.DesktopGameSource ();
-		dummy_source.each_game ((game) => {
+		if (connection == null)
+			return;
```

Comment:
Move to `TrackerGameSource`, so only tracker-found games are impacted?
Other game sources may work fine...